utils/etherscan.ts

```diff
@@ -0,0 +1,27 @@
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+
+export const sleep = (ms: number): Promise<void> =>
+  new Promise((resolve) => setTimeout(resolve, ms))
+
+interface VerifyEtherscan {
+  address: string
+  contract?: string
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+  constructorArguments?: any[]
+  libraries?: {
+    [libraryName: string]: string
+  }
+}
+
+export const verifyEtherscan = async (
```

Comment:
I would imagine there is a library for this out there in the wild

Reply:
using the @nomiclabs/hardhat-etherscan, this file is pulled from our current contracts repo